rmgpy/data/kinetics/database.py

```diff
@@ -107,86 +107,101 @@ def load(self, path, families=None, libraries=None, depositories=None):
+        selection will be inverted to families NOT in the list.
         """
-        
-        familiesToLoad = []
         for (root, dirs, files) in os.walk(os.path.join(path)):
             if root == path:
                 break  # all we wanted was the list of dirs in the base path
 
-        if families == 'default':
-            logging.info('Loading default kinetics families from {0}'.format(path))
-            for d in dirs:  # load them in directory listing order, like other methods (better than a random dict order)
-                try:
-                    recommended = self.recommendedFamilies[d]
-                except KeyError:
-                    raise DatabaseError('Family {0} not found in recommendation list (probably at {1}/recommended.py)'.format(d, path))
-                if recommended:
-                    familiesToLoad.append(d)
-            for label, value in self.recommendedFamilies.iteritems():
-                if label not in dirs:
-                    raise DatabaseError('Family {0} found (in {1}/recommended.py) not found on disk.'.format(label, path))
-
-        elif families == 'all':
-            # All families are loaded
-            logging.info('Loading all of the kinetics families from {0}'.format(path))
-            for d in dirs:
-                familiesToLoad.append(d)
-        elif families == 'none':
-            logging.info('Not loading any of the kinetics families from {0}'.format(path))
-            # Don't load any of the families
-            familiesToLoad = []
-        elif isinstance(families, list):
-            logging.info('Loading the user-specified kinetics families from {0}'.format(path))
-            # If all items in the list start with !, all families will be loaded except these
-            if len(families) == 0:
-                raise DatabaseError('Kinetics families should be a non-empty list, or set to `default`, `all`, or `none`.')
-            elif all([label.startswith('!') for label in families]):
-                for d in dirs:
-                    if '!{0}'.format(d) not in families:
-                        familiesToLoad.append(d)
-            elif any([label.startswith('!') for label in families]):
-                raise DatabaseError('Families list must either all or none have prefix "!", but not a mix.')
-            else:  # only the families given will be loaded
-                for d in dirs:
-                    if d in families:
-                        familiesToLoad.append(d)
-                for label in families:
-                    if label not in dirs:
-                        raise DatabaseError('Family {0} not found on disk.'.format(label))
+        all_families = set(dirs)
+
+        # Convert input to a list to simplify processing
+        if not isinstance(families, list):
+            families = [families]
+
+        # Check for ! syntax, which allows specification of undesired families
+        if all([item.startswith('!') for item in families]):
```

Comment:
You can remove the square brackets around `item.startswith('!') for item in families`, because `all` accepts generator expressions.

Reply:
Thanks, fixed.